src/controls/Ft3asTemplateSelector.tsx

```diff
@@ -1,25 +1,48 @@
 import { DefaultButton, Dropdown, IDropdownOption, Panel, PrimaryButton } from "@fluentui/react";
-import { FormEvent, useCallback, useState } from "react";
+import { FormEvent, useCallback, useEffect, useState } from "react";
+import TemplateServiceInstance from "../service/TemplateService";
 
 const buttonStyles = { root: { marginRight: 8 } };
 
 interface Ft3AsTemplateSelectorProps {
     isOpen: boolean;
-    availableTemplates: string[];
     onTemplateSelected: (templateUrl?: string) => void;
     onClose: () => void;
 }
+
 export default function Ft3AsTemplateSelector(props: Ft3AsTemplateSelectorProps) {
     const { isOpen } = props;
-    // const [availableTemplates, setAvailableTemplates] = useState<IDropdownOption[]>([]);
-    const availableTemplates = props.availableTemplates.map<IDropdownOption>(t => {
-        return {
-            key: t, text: t
+    const [availableTechnologies, setAvailableTechnologies] = useState<IDropdownOption[]>([]);
+    const [availableLanguages, setAvailableLanguages] = useState<IDropdownOption[]>([]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            await TemplateServiceInstance.init();
+            const technologyNames = TemplateServiceInstance.getAvailableTemplateNames();
+            setAvailableTechnologies(technologyNames.map<IDropdownOption>(t => {
+                return {
+                    key: t, text: t
+                }
+            }));
         }
-    });
-    const [selectedItem, setSelectedItem] = useState<IDropdownOption>();
+        fetchData()
+            .then(() => console.log('loaded'))
+            .catch(reason => {
+                console.error(reason);
+            })
+    }, []);
+
+    const [selectedTechItem, setSelectedTechItem] = useState<IDropdownOption>();
+    const [selectedLanguageItem, setSelectedLanguageItem] = useState<IDropdownOption>();
+
     const onOk = () => {
-        props.onTemplateSelected(selectedItem?.key as string | undefined);
+        if (selectedLanguageItem && selectedTechItem){
+            console.log(selectedTechItem.text);
+            console.log(selectedLanguageItem.text);
+            var url : string = TemplateServiceInstance.getPathforTechAndLanguage(selectedTechItem.text, selectedLanguageItem.text);
```

Comment:
I would replace _var url_ by _const url_ or _let url_ if it is mutable. The behavior of var in javascript is very unpredictable as it behaves as a global variable.

Reply:
fixed.